src/Microsoft.Kiota.Abstractions.csproj

```diff
@@ -6,20 +6,23 @@
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
     <AssemblyTitle>Kiota Abstractions Library for dotnet</AssemblyTitle>
     <Authors>Microsoft</Authors>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;net6.0;</TargetFrameworks>
```

Comment:
would adding netstandard2.1 not be enough to get the functionality? it'd also git us a broader compatibility range.

Reply:
Hey @baywet,

My understanding is that the Nullable Reference Types feature is only a compile time feature and does not really create new types but annotations on the existing reference types. The consumer of the library would then opt in/out of the warning/errors by specifying the relevant tag in the `.csproj` file. This implies that we could still distribute the library with only the `netstandard2.0` target and the nullable information would still be propagated if one opted in. (I've also verified with existing libraries like Azure.Core which [enable NRT](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Azure.Core.csproj) and still distribute [netstandard2.0 target](https://www.nuget.org/packages/Azure.Core))

The real reason for the extra target is really for OUR developer experience in order to have accurate nullable errors and context so that we do not propagate inaccurate nullable information as updating the language version in the project will indeed enable the feature but is only well supported in the later target frameworks. 

With this in mind, think `net6.0` a better target as `netstandard2.1` doesn't really add anything as `netstandard2.0` covers all that `netstandard2.1` has. In the event new language features are added, I suspect that they will most likely be built on top of `net6.0`([or its future equivalent?](https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/)) rather than netstandard which also creates more options/benefits for everyone and net framework will be covered through the `netstandard2.0` target.

Reply:
Thanks for detailing things out here. I was under the impression that some of the NRT feature required runtime support, hence our compatibility conundrum. But this is probably a requirement only when you're using the attributes associated with this feature (as opposed to the ? ! symbols).
As per the target framework. The lower the version, the broader the compatibility base. Azure core, [is targeting standard 2.1](https://github.com/Azure/azure-sdk-for-net/blob/124a959609df8999f6e08265af35590a111ea6ef/sdk/core/Azure.Core/src/Azure.Core.csproj#L11).
I believe we don't need anything specific from net5 or net6 targets, [this page](https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0) gives a great overview of the topic. The important part being at the very bottom, by targeting net5/6, we'd be dropping net core 3.1, mono, and xamarin, which I believe we still want to support.

Reply:
Resolved in latest updates